### ISSUE TITLE
CS8618 should not be generated on constructors that throw if a property is null

### ISSUE THREAD

**Issue by Foxite** [dotnet/roslyn#39740]
**Version Used**: 3.3.1-beta3-19461-02 (2fd12c21)

```csharp
#nullable enable
public class TestClass {
	public string TestString { get; private set; }
	
	public TestClass() {
		// Some code that may or may not initialize TestString
		// For example a foreach that looks for the right item to use for TestString
		
		if (TestString == null) {
			throw new Exception();
		}
	}
}
```

**Expected Behavior**: No warning is emitted as there is no way for the constructor to return normally without having initialized `TestString`.

**Actual Behavior**: A CS8618 (Non-nullable property is uninitialized) warning is emitted on the constructor.

**Comment by Foxite** [comment]
This has been previously reported by #25529 and fixed by #34316, but I can't tell if it's been released yet.

**Comment by Happypig375** [comment]
Bump! Hit this today.
